var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -48,67 +51,198 @@ class Hypre(Package):
     variant('shared', default=(sys.platform != 'darwin'),
             description="Build shared library (disables static library)")
     # SuperluDist have conflicting headers with those in Hypre
-    variant('internal-superlu', default=True,
+    variant('internal-superlu', default=False,
             description="Use internal Superlu routines")
     variant('int64', default=False,
             description="Use 64bit integers")
     variant('mpi', default=True, description='Enable MPI support')
-
-    # Patch to add ppc64le in config.guess
-    patch('ibm-ppc64le.patch', when='@:2.11.1')
+    variant('fei', default=False, description='Use internal FEI routines')
+    variant('mli', default=False, description='Use MLI')
+    variant('cuda', default=False,
+            description='Use CUDA. Require cuda 8.0 or higher')
+    variant('openmp', default=False, description='Use OpenMP')
+    variant('complex', default=False, description='Use complex values')
+    # TODO: Add 'cuda_arch' variant
```

Comment:
if you derive additionally from `CudaPackage`, you will get the `cuda`, `cuda_arch` and whole lot of constraints on compilers / cuda_arch combo for free.

Reply:
see https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cuda.py

Reply:
Ha, nice, I did not know about that package type. I'll use that. Thanks for the tip. :+1: